### ISSUE TITLE
react-aria useDatePicker has two elements with the same id

### ISSUE THREAD

**Issue by JennyHumphries** [adobe/react-spectrum#3969]
### Expected Behaviour
react-aria useDatePicker should be compliant with accessibility rules

### Actual Behaviour
Following the example in the docs here: https://react-spectrum.adobe.com/react-aria/useDatePicker.html
Both `groupProps` and `fieldProps` have the same `id` which results in elements with duplicate ids.

#### Steps to Reproduce
In the example docs here: https://react-spectrum.adobe.com/react-aria/useDatePicker.html
If you inspect the example date picker you can see the duplicate ids here:

![image](https://user-images.githubusercontent.com/8667404/215148335-d2073721-b678-4bd8-a886-8b062a749342.png)

#### Platform and Version
react-aria ^3.21.0

#### Current workaround
Manually set the id on the groupProps to something different

**Comment by LFDanLu** [comment]
Thanks for reporting this, at a glance maybe https://github.com/adobe/react-spectrum/blob/f32367ad4a4639ccb8738b9daefe42d0d870affd/packages/%40react-aria/datepicker/src/useDatePicker.ts#L115 shouldn't be added here, but I'd have to double check the accessibility experience

**Comment by dannify** [comment]
The div with role="presentation" shouldn't be getting the id as well.
